frontend/src/app/components/mempool-graph/mempool-graph.component.ts

```diff
@@ -304,12 +446,47 @@ export class MempoolGraphComponent implements OnInit, OnChanges {
           type: 'category',
           boundaryGap: false,
           axisLine: { onZero: true },
+          axisTick: {
+            lineStyle: {
+              width: 0,
+            },
+          },
           axisLabel: {
+            interval: labelDates.length ? labelDates.length : 'auto',
             align: 'center',
             fontSize: 11,
-            lineHeight: 12,
+            lineHeight: 25,
+            customValues: labelDates.length ? labelDates : null,
+            formatter: (value: string, index: number) => {
+              const date = new Date(value);
+              if (this.template === 'widget') {
+                return date.toLocaleTimeString(this.locale, { hour: 'numeric', minute: 'numeric' });
+              }
+
+              // first date
+              if (index === 0) {
+                return date.toLocaleDateString(this.locale, {  year: '2-digit', month: 'short', 'day': 'numeric' });
+              }
+
+              switch (this.windowPreference) {
+                case '2h':
+                  return date.toLocaleTimeString(this.locale, { hour: 'numeric', minute: 'numeric' });
+                case '24h':
+                  return date.toLocaleTimeString(this.locale, { hour: 'numeric' });
+                case '1w':
+                  return date.toLocaleDateString(this.locale, { month: 'short', day: 'numeric', weekday: 'short' });
+                case '1m':
+                case '3m':
+                case '6m':
+                case '1y':
+                  return date.toLocaleDateString(this.locale, { month: 'short', day: 'numeric' });
```

Comment:
It feels like the syntaxes you were using in PR 900 looked nicer for the 6M chart, could you explain why you changed it to these instead? I'm not sure we want to use the same syntax for these 4 timespans